### PR TITLE
Update telemetry documentation and metric names/descriptions

### PR DESCRIPTION
**Pull request by ljoss17**
Closes: #2479

## Description

The following metrics have been renamed:

* `ibc_client_updates` to `client_updates_submitted`
* `ibc_client_misbehaviours ` to `client_misbehaviours_submitted`
* `ibc_receive_packets` to `receive_packets_confirmed`
* `ibc_acknowledgment_packets ` to `acknowledgment_packets_confirmed`
* `ibc_timeout_packets ` to `timeout_packets_confirmed`
* `cache_hits` to `queries_cache_hits`
* `msg_num` to `total_messages_submitted`
* `send_packet_count` to `send_packet_events`
* `acknowledgement_count` to `acknowledgement_events`
* `cleared_send_packet_count` to `cleared_send_packet_events`
* `cleared_acknowledgment_count` to `cleared_acknowledgment_events`

The following metrics have been added:

* `timeout_events`

Fixed the following metrics:

* `client_updates_submitted`: correctly count all ClientUpdate messages
* `total_messages_submitted`: count only submitted messages

### Testing

__Timeout events__

Use standard Hermes `config.toml`, for example:

<details>
<summary>config.toml</summary>

```
[global]
log_level = 'info'

[mode]

[mode.clients]
enabled = true
refresh = true
misbehaviour = true

[mode.connections]
enabled = true

[mode.channels]
enabled = true

[mode.packets]
enabled = true
clear_interval = 100
clear_on_start = true
tx_confirmation = true

[telemetry]
enabled = true
host = '127.0.0.1'
port = 3001

[[chains]]
id = 'ibc-0'
rpc_addr = 'http://localhost:27000'
grpc_addr = 'http://localhost:27002'
websocket_addr = 'ws://localhost:27000/websocket'
rpc_timeout = '15s'
account_prefix = 'cosmos'
key_name = 'wallet'
store_prefix = 'ibc'
gas_price = { price = 0.01, denom = 'stake' }
max_gas = 10000000
clock_drift = '5s'
trusting_period = '14days'
trust_threshold = { numerator = '1', denominator = '3' }

[[chains]]
id = 'ibc-1'
rpc_addr = 'http://localhost:27010'
grpc_addr = 'http://localhost:27012'
websocket_addr = 'ws://localhost:27010/websocket'
rpc_timeout = '15s'
account_prefix = 'cosmos'
key_name = 'wallet'
store_prefix = 'ibc'
gas_price = { price = 0.01, denom = 'stake' }
max_gas = 10000000
clock_drift = '5s'
trusting_period = '14days'
trust_threshold = { numerator = '1', denominator = '3' }
```

</details>

Setup:

1. `gm start`
2. `gm hermes keys`
3. `hermes create channel --a-chain ibc-0 --b-chain ibc-1 --a-port transfer --b-port transfer --new-client-connection --yes`

Test:

1. `hermes start`
2. `hermes tx ft-transfer --dst-chain ibc-0 --src-chain ibc-1 --src-port transfer --src-channel channel-0 --amount 100 --timeout-seconds 120`
3. `hermes tx ft-transfer --dst-chain ibc-0 --src-chain ibc-1 --src-port transfer --src-channel channel-0 --amount 100 --timeout-seconds 3`

The first transfer should be successful and the second should timeout, so the metrics should be:

```
acknowledgement_events{chain="ibc-1",channel="channel-0",counterparty="ibc-0",port="transfer"} 1
send_packet_events{chain="ibc-1",channel="channel-0",counterparty="ibc-0",port="transfer"} 2
timeout_events{chain="ibc-0",channel="channel-0",counterparty="ibc-1",port="transfer"} 1
```

__Client update metric and total messages__

Update Hermes `config.toml` to have `trusted period = '2minutes'` for chain `ibc-0`, or use the following `config.toml`:

<details>
<summary>config.toml</summary>

```
[global]
log_level = 'info'

[mode]

[mode.clients]
enabled = true
refresh = true
misbehaviour = true

[mode.connections]
enabled = true

[mode.channels]
enabled = true

[mode.packets]
enabled = true
clear_interval = 100
clear_on_start = true
tx_confirmation = true

[telemetry]
enabled = true
host = '127.0.0.1'
port = 3001

[[chains]]
id = 'ibc-0'
rpc_addr = 'http://localhost:27000'
grpc_addr = 'http://localhost:27002'
websocket_addr = 'ws://localhost:27000/websocket'
rpc_timeout = '15s'
account_prefix = 'cosmos'
key_name = 'wallet'
store_prefix = 'ibc'
gas_price = { price = 0.01, denom = 'stake' }
max_gas = 10000000
clock_drift = '5s'
trusting_period = '2minutes'
trust_threshold = { numerator = '1', denominator = '3' }

[[chains]]
id = 'ibc-1'
rpc_addr = 'http://localhost:27010'
grpc_addr = 'http://localhost:27012'
websocket_addr = 'ws://localhost:27010/websocket'
rpc_timeout = '15s'
account_prefix = 'cosmos'
key_name = 'wallet'
store_prefix = 'ibc'
gas_price = { price = 0.01, denom = 'stake' }
max_gas = 10000000
clock_drift = '5s'
trusting_period = '14days'
trust_threshold = { numerator = '1', denominator = '3' }
```

</details>

Setup:

1. `gm start`
5. `gm hermes keys`
6. `hermes create channel --a-chain ibc-0 --b-chain ibc-1 --a-port transfer --b-port transfer --new-client-connection --yes`

Test:

1. `hermes start`
2. wait for 1 refresh, this is shown by the output `INFO ThreadId(30) refresh{client=07-tendermint-0 src_chain=ibc-0 dst_chain=ibc-1}: [ibc-0 -> ibc-1:07-tendermint-0] client requires refresh`
3. Confirm on [http://localhost:3001/metrics](http://localhost:3001/metrics) that there has been 1 `ClientUpdate` message: `client_updates_submitted{chain="ibc-1",client="07-tendermint-0"} 1` and 1 total message for chain `ibc-1`: `total_messages_submitted{chain="ibc-1"} 1`
7. `hermes tx conn-init --dst-chain ibc-0 --src-chain ibc-1 --dst-client 07-tendermint-0 --src-client 07-tendermint-0`
8. Wait for the connection to be completely created, this is shown by the output: `INFO ThreadId(228) connection{connection=connection::connection-1:ibc-0->ibc-1}: connection handshake step completed with events: OpenConfirmConnection(...`

The instance of `hermes start` has finished opening the connection by sending the following messages:
1. `ClientUpdate` to `ibc-0`
2. `ClientUpdate` + `OpenTryConnection` to `ibc-1`
3. `ClientUpdate` to `ibc-1`
4. `ClientUpdate` + `OpenAckConnection` to `ibc-0`
5. `ClientUpdate` + `OpenAckConnection` to `ibc-1`

Taking into account the ClientUpdate from the refresh the metrics should be:

```
client_updates_submitted{chain="ibc-0",client="07-tendermint-0"} 2
client_updates_submitted{chain="ibc-1",client="07-tendermint-0"} 4

total_messages_submitted{chain="ibc-0"} 3
total_messages_submitted{chain="ibc-1"} 6
```

______

### PR author checklist:
- [x] Added changelog entry, using [`unclog`](https://github.com/informalsystems/unclog).
- [ ] ~Added tests: integration (for Hermes) or unit/mock tests (for modules).~
- [x] Linked to GitHub issue.
- [x] Updated code comments and documentation (e.g., `docs/`).
- [ ] Tagged *one* reviewer who will be the one responsible for shepherding this PR.

### Reviewer checklist:

- [ ] Reviewed `Files changed` in the GitHub PR explorer.
- [ ] Manually tested (in case integration/unit/mock tests are absent).

### CODE REVIEW

File: guide/src/telemetry/operators.md [hunk 1]
```diff
@@ -0,0 +1,100 @@
...
+
+This section is a basic guide on how Hermes metrics can be used to observe both
+the current state of the Hermes relayer and the networks it is connected to.
+
+## General remarks about the metrics
+
+- All Hermes metrics are tracked and updated from the moment the Hermes service (i.e., `start`) starts up.
+Metrics are automatically reset if the service is restarted.
+- For maximum reliability, it is advised to combine monitoring of your Hermes service with monitoring of your full nodes.
+- TODO: Grafana template here.
```
Comment (adizere) [inline guide/src/telemetry/operators.md:11]: FYI @AlianBenabdallah this is likely where we're reference a Grafana template.

**Review by romac — APPROVED** [review]
Amazing!

File: guide/src/telemetry/operators.md (hunk 1 above)
Comment (AlianBenabdallah) [reply guide/src/telemetry/operators.md]: I think we can add Grafana templates in another PR

**Comment by AlianBenabdallah** [comment]
Should we include discovery dependencies in this PR ? @ljoss17

**Comment by ljoss17** [comment]
> Should we include discovery dependencies in this PR ? @ljoss17

As we usually update the guide in the same PR where the feature is added/updated, I think it is best to update the guide with the dependencies in the PR introducing them, https://github.com/informalsystems/ibc-rs/pull/2505.

**Review by adizere — APPROVED** [review]
Awesome work, LGTM!

**Pull request merged**